File: ex03/options.pl
#!/usr/bin/env perl

use v5.30;
use utf8;

use Getopt::Long;

my $options = {};
GetOptions($options, qw(alpha|a beta|b gamma|g=s))
    or exit 1;

say "OPTIONS";
foreach my $key (sort keys %$options) {
    say " * $key is set to value $options->{$key}";
}

say "ARGUMENTS";
foreach my $arg (@ARGV) {
    say " - $arg";
}

File: ex03/options.py
#!/usr/bin/env python3

import getopt
import sys

opts, args = getopt.gnu_getopt(sys.argv[1:], "abg:", ["alpha", "beta", "gamma="])

print("OPTIONS")
for opt in opts:
    print(" *", opt[0], opt[1])

print("ARGUMENTS")
for arg in args:
    print(" *", arg)
